### PR TITLE
Generate both vertical and horizontal mouse wheel events for diagonal scrolling

### PR DESCRIPTION
**Pull request by vadz**
See #24516 for motivation.

@craftyjon Please let me know if this works for you.

@csomor Please let me know if you see a better way to translate a single `NSEvent` to multiple `wxMouseEvent`s than what I did here.

### CODE REVIEW

**Comment by csomor** [comment]
@vadz thanks, that's something that would probably work without changes needed in the client apps, I was thinking whether we'd need some (new?) 'panning' event having two dimensions to avoid Manhattan geometry movements ...

File: src/osx/cocoa/window.mm [hunk 1]
```diff
@@ -3933,9 +3955,18 @@ new wxCocoaGesturesImpl(this, m_osxView, eventsMask)
...
     (void)SetupCursor(event);
 
-    wxMouseEvent wxevent(wxEVT_LEFT_DOWN);
-    SetupMouseEvent(wxevent , event) ;
-    return GetWXPeer()->HandleWindowEvent(wxevent);
+    bool processed = false;
+    for ( auto& wxevent : TranslateMouseEvent(event) )
+    {
+        // Even if this event was processed, still continue with the other
+        // eveents, if any.
```
Comment (dsa-t) [inline src/osx/cocoa/window.mm:3962]: Typo.
```suggestion
        // events, if any.
```

File: src/osx/cocoa/window.mm [hunk 2]
```diff
@@ -787,17 +791,28 @@ - (CGFloat)scrollingDeltaY;
             wxevent.m_linesPerAction = 1;
             wxevent.m_columnsPerAction = 1;
 
-            if ( fabs(deltaX) > fabs(deltaY) )
+            // Should we impose some minimum threshhold here?
```
Comment (dsa-t) [inline src/osx/cocoa/window.mm:794]:
```suggestion
            // Should we impose some minimum threshold here?
```

File: src/osx/cocoa/window.mm (hunk 1 above)
Comment (vadz) [reply src/osx/cocoa/window.mm:3962]: Thanks, I'll fix this (and the other) typo(s) when I merge it, if it works.

**Comment by craftyjon** [comment]
@vadz This works well out of the box!  No changes needed on our side.  Thank you :)

**Pull request merged**